blueocean-github-pipeline/src/test/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubOrgFolderPermissionsTest.java

```diff
@@ -0,0 +1,166 @@
+import org.jvnet.hudson.test.TestExtension;
+
+import com.cloudbees.hudson.plugins.folder.Folder;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import hudson.model.TopLevelItem;
+import hudson.model.User;
+import io.jenkins.blueocean.rest.factory.organization.OrganizationFactory;
+import io.jenkins.blueocean.rest.model.BlueOrganization;
+import io.jenkins.blueocean.service.embedded.OrganizationFactoryImpl;
+import io.jenkins.blueocean.service.embedded.rest.OrganizationImpl;
+import jenkins.branch.OrganizationFolder;
+import jenkins.model.Jenkins;
+import jenkins.model.ModifiableTopLevelItemGroup;
+
+
+public class GithubOrgFolderPermissionsTest extends GithubMockBase {
+
+    @Test
+    public void canCreateWhenHavePermissionsOnDefaultOrg() throws Exception {
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Jenkins.ADMINISTER).everywhere().to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(true);
+    }
+
+    @Test
+    public void canNotCreateWhenHaveNoPermissionOnDefaultOrg() throws Exception {        
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Item.READ).everywhere().to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(false);
+    }
+
+    @Test
+    public void canCreateWhenHavePermissionsOnCustomOrg() throws Exception {
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Item.READ).everywhere().to(user);
+        authz.grant(Item.CREATE, Item.CONFIGURE).onFolders(getOrgRoot()).to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(true);
+    }
+
+    @Test
+    public void canNotCreateWhenHaveNoPermissionOnCustomOrg() throws Exception {
+        MockAuthorizationStrategy authz = new MockAuthorizationStrategy();
+        authz.grant(Item.READ).everywhere().to(user);
+        j.jenkins.setAuthorizationStrategy(authz);
+        // refresh the JWT token otherwise all hell breaks loose.
+        jwtToken = getJwtToken(j.jenkins, "vivek", "vivek");
+        createGithubOrgFolder(false);
+    }
+
+    private void createGithubOrgFolder(boolean shouldSuceed) throws Exception {
+        String credentialId = createGithubCredential(user);
+        String orgFolderName = "cloudbeers1";
+        Map resp = new RequestBuilder(baseUrl)
```

Comment:
@vivek are you saying that creating the GH Org is in a full blown Jenkins instance not synchronous?  (surely creating the Item must be synchronous with the request in every case - otherwise what if 2 users tried to create an org with the same name - both would be told it was created, yet one of them would loose)?  (confused?(